pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -646,13 +707,18 @@ func (c *Controller) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(acReconciler)
 }
 
-func GetProvisioningRequestName(workloadName, checkName string) string {
-	fullName := fmt.Sprintf("%s-%s", workloadName, checkName)
+func GetProvisioningRequestName(workloadName, checkName string, attempt int32) string {
+	fullName := fmt.Sprintf("%s-%s-%d", workloadName, checkName, int(attempt))
 	return limitObjectName(fullName)
 }
```

Comment:
@mimowo @alculquicondor What happens if admins upgrade kueue v0.4 to the latest version?
I guess that ProvisioningRequests (PR) using the old name form will be recreated, and I'm wondering if this name form change will affect the PRs if CA is provisioning nodes triggered by the PRs.

Reply:
Regarding PRs with `CapacityAvailable`, `Provisioned`, or `Failed`, I don't think that the above concerns.

Reply:
It will be considered attempt 1.

Reply:
> It will be considered attempt 1.

You mean that PRs with old name forms wouldn't replaced, right? If so, that makes sense.
Thanks!

Reply:
Actually, the old PRs need to be recreated as they would not pass the [`matches` function](https://github.com/kubernetes-sigs/kueue/blob/92079c0c391ba74a5e09c0d48113b33619f00a18/pkg/controller/admissionchecks/provisioning/controller.go#L735).

It is easy to match old PRs (without the suffix) as attempt "1", but one thing to consider is that provisioning request `wl-ac-1`, would be matching to both `ac` and `ac-1`. 

I think this is ok to leave as is, because there is no support for provisioning requests in Kueue 0.4, it starts in 0.5.0 and the `ProvisioningACC` is [Alpha](https://github.com/kubernetes-sigs/kueue/blob/92079c0c391ba74a5e09c0d48113b33619f00a18/pkg/features/kube_features.go#L52-L57). See [feature-stages](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-stages). 

@mwielgus @alculquicondor @trasc  WDYT?

Reply:
Added a release note.

Reply:
It is always good to have additional safety and not loosing any in-flight requests, but additional work to perfect upgrade story for just released non-default, alpha-like feature seem to have relatively little impact and return on investment.

Reply:
> It is always good to have additional safety and not loosing any in-flight requests, but additional work to perfect upgrade story for just released non-default, alpha-like feature seem to have relatively little impact and return on investment.

That makes sense. I'm ok with leaving on recreation for ProvisioningRequest since the ProvisioningACC is the Alpha stage.

I believe that users don't use the featureGate yet in production.

Reply:
@mimowo Thanks for updating the release note. The notification would be helpful :)